File: app/src/main/java/com/speaktool/base/BaseFragmentActivity.java
package com.speaktool.base;

import android.content.Context;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

import com.speaktool.R;

/**
 * @author maple
 * @time 2018/10/11
 */
public abstract class BaseFragmentActivity extends FragmentActivity {
    public Context mContext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);// 保持竖屏
        mContext = getBaseContext();
    }

    @Override
    public void onBackPressed() {
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fl_content);
        if (fragment instanceof BaseFragment) {
            if (!((BaseFragment) fragment).onKeyBackPressed()) {
                backFragment();
            }
        } else {
            super.onBackPressed();
        }
    }

    // ------------------ fragment ------------------

    public void addView(Fragment fgView) {
        addView(fgView, R.id.fl_content);
    }

    public void addView(Fragment fgView, int containerViewId) {
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.add(containerViewId, fgView).commit();
    }

    public void replaceView(Fragment fgView) {
        replaceView(fgView, R.id.fl_content);
    }

    public void replaceView(Fragment fgView, int containerViewId) {
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(containerViewId, fgView);
        ft.addToBackStack(null);
        ft.commit();
    }

    public void backFragment() {
        if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
            getSupportFragmentManager().popBackStack();
        } else {
            finish();
        }
    }

//    public boolean onNext() {
//        int nextPage = getSupportFragmentManager().getBackStackEntryCount() + 1;
//        if (fragmentList != null && fragmentList.size() > nextPage) {
//            replaceView(fragmentList.get(nextPage));
//            return true;
//        }
//        return false;
//    }

}

File: app/src/main/java/com/speaktool/base/BaseFragmentActivity.kt
package com.speaktool.base

import android.content.Context
import android.content.pm.ActivityInfo
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.speaktool.R

/**
 * @author maple
 * @time 2018/10/11
 */
abstract class BaseFragmentActivity : FragmentActivity() {
    lateinit var mContext: Context

    override
    fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT// 保持竖屏
        mContext = this
    }

    override
    fun onBackPressed() {
        val fragment = supportFragmentManager.findFragmentById(R.id.fl_content)
        if (fragment is BaseFragment) {
            if (!fragment.onKeyBackPressed()) {
                backFragment()
            }
        } else {
            super.onBackPressed()
        }
    }

    // ------------------ fragment method ------------------

    @JvmOverloads
    fun addView(fgView: Fragment, containerViewId: Int = R.id.fl_content) {
        val ft = supportFragmentManager.beginTransaction()
        ft.add(containerViewId, fgView).commit()
    }

    @JvmOverloads
    fun replaceView(fgView: Fragment, containerViewId: Int = R.id.fl_content) {
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(containerViewId, fgView)
        ft.addToBackStack(null)
        ft.commit()
    }

    private fun backFragment() {
        if (supportFragmentManager.backStackEntryCount > 0) {
            supportFragmentManager.popBackStack()
        } else {
            finish()
        }
    }

    //    public boolean onNext() {
    //        int nextPage = getSupportFragmentManager().getBackStackEntryCount() + 1;
    //        if (fragmentList != null && fragmentList.size() > nextPage) {
    //            replaceView(fragmentList.get(nextPage));
    //            return true;
    //        }
    //        return false;
    //    }

}